### ISSUE TITLE
Unflatten?

### ISSUE THREAD

**Issue by dufferzafar** [nickjer/fltn#2]
It should be possible to unflatten the individual rows back into a JSON.

https://github.com/tomnomnom/gron has `--ungron` but it is JSON only, while fltn supports CSV as well!

**Comment by nickjer** [comment]
Yeah, I purposely didn't tackle "un-flattening" because it was a use case I never needed. It would also be a much larger lift as I would be responsible for writing a custom de-serializer for the output.

I see two possibilities (although there may be more) for tackling this issue:
- The format of the output should be the same as `gron` so theoretically you could just pipe this to `gron --ungron` to "un-flatten" (I never tested this, so take this with a grain of salt)
- Someone else could tackle writing a deserializer for this output (whether in a custom library or in this code), and I could try to get it incorporated in

Note, my output should be nearly identical to gron's but I did take some liberties for instance using JSON to serialize the object keys in the output. It just seemed safer and more consistent, for instance it handled more edge cases with strange unicode than `gron` did.
